packages/paste-website/src/components/tokens-list/index.tsx

```diff
@@ -1,136 +1,88 @@
 import * as React from 'react';
+
+import debounce from 'lodash/debounce';
+import {useUIDSeed} from '@twilio-paste/uid-library';
 import {Box} from '@twilio-paste/box';
-import {Stack} from '@twilio-paste/stack';
 import {Label} from '@twilio-paste/label';
-import {Heading} from '@twilio-paste/heading';
-import {Card} from '@twilio-paste/card';
-import {Button} from '@twilio-paste/button';
 import {Input} from '@twilio-paste/input';
-import type {ThemeShape} from '@twilio-paste/theme';
-import {Table, Tr, Th, Td, THead, TBody} from '@twilio-paste/table';
-import Tokens from '@twilio-paste/design-tokens/dist/tokens.gatsby.js';
-import {Text} from '@twilio-paste/text';
-import {useUID} from '@twilio-paste/uid-library';
-import {InlineCode} from '../Typography';
-import {AnchoredHeading} from '../Heading';
-import {TokenExample} from './TokensExample';
-import {getTokenValue} from './getTokenValue';
+import {FilterIcon} from '@twilio-paste/icons/esm/FilterIcon';
+
+import {CategorySection} from './category-section';
 import {useDarkModeContext} from '../../context/DarkModeContext';
-import {trackTokenFilterString, filterTokenList, getTokensByTheme} from './helpers';
-import type {Token, TokenCategory, TokensListProps} from './types';
+import {trackTokenFilterString} from './utils';
+
 import {PageAside} from '../shortcodes/PageAside';
 import {NoTokensFound} from './NoTokensFound';
-import {BackgroundColor} from './token-box/BackgroundColor';
-import {NoResultImage} from '../images/EmptyStateImages';
-import {TokenCard} from './token-card';
-import {FilterIcon} from '@twilio-paste/icons/esm/FilterIcon';
+import {TOKEN_CATEGORIES, pageAsideData} from './constants';
 
-const PreviewComponentMap: {[key: string]: React.ReactNode} = {
-  'background-colors': BackgroundColor,
-};
+const ContentWrapper: React.FC = (props) => <Box as="div" display={['block', 'block', 'flex']} {...props} />;
+const Content: React.FC = (props) => <Box as="div" maxWidth="size70" minWidth="0" {...props} />;
 
-console.log(Tokens);
-const TokenKeys = Object.keys(Tokens);
+// @TODO add sorting.
+// const collator = new Intl.Collator(undefined, {numeric: true, sensitivity: 'base'});
+export const TokensList: React.FC = () => {
+  const seed = useUIDSeed();
+  const {theme} = useDarkModeContext();
 
-const ContentWrapper = (props): React.FC => <Box as="div" display={['block', 'block', 'flex']} {...props} />;
-const Content = (props): React.FC => <Box as="div" maxWidth="size70" minWidth="0" {...props} />;
+  const [filterString, setFilterString] = React.useState('');
+  const [noResults, setNoResults] = React.useState(0);
 
-const sentenceCase = (catName: string): string => {
-  return catName
-    .split('-')
-    .join(' ')
-    .replace(/[a-z]/i, (letter): string => {
-      return letter.toUpperCase();
-    });
-};
+  const handleSetNoResults = React.useCallback(() => setNoResults((total) => total + 1), []);
```

Comment:
Note: memoizing this since it is being passed to multiple children categories that are getting mapped over. The  categories are somewhat static, but we don't want this to trigger re-renders.